### DISCUSSION TITLE
Project: G7 receiver via USB to clarity, automated via Raspberry PI or Android device?

### DISCUSSION THREAD

**Discussion by pdg212** [DiaKEM/dexcom-g7-apk-patcher#33] [Q&A]
Hello!  Our daughter recently moved to Dexcom G7 and the size, warm-up time and accuracy are all great... but the signal strength to phone is AWFUL.  Tried several devices and the phone can be right next to her but not pick up a reading - very, very frustrating.

The official Dexcom G7 receiver, however, has very good range - certainly acceptable enough and sufficient to simply be in the same room.  Unfortunately it's a 'dumb' reader - and cannot transmit readings to Dexcom follow or similar.

The reader does have a USB download capability however, and it's possible to connect it to a computer and upload the data to a PC/Dexcom Clarity.

The ask here is whether anyone can create a simple 'uploader' device that can be essentially strapped to the Dexcom reader, plugged in to the USB port, extract the data and upload to clarity every 5-10 minutes, so we can remotely monitor BG readings from home with some degree of confidence.  This could perhaps be a Raspberry PI or an Android device.  Would need to have SIM so data can be uploaded remotely without any reliance on a WIFI connection.

This would be a lifesaver.  At home we can use the G7 receiver... however at school we have very little faith that the teachers will monitor and act on BG levels dropping.  We'd feel much happier if there's a way of seeing her G7 data remotely, but connectivity from sensor to phone directly is not strong enough, so hoping there can be a creative guru who is able to automatically extract from the G7 receiver directly instead (where the signal reception is good).

**Comment by khskekec** [comment]
Hi @pdg212,

Unfortunately we had the same problem and tried it with several phones with small dimensions because my son is 6 years old. A big phone is no option to us.

We finally switched to King Kong 3 mini and all problems were gone. The smartphone before, king Kong mini 2 was although terrible. Connection loss and nearly 70 seconds to send a bolus over AAPS by Bluetooth to omnipod dash.

The mini 3 solved all this problems, it is amazing.

Your approach is really interesting. To be honest my knowledge here is limited so I cannot promise you some solution here.

The only advice I can give you is to try out a smartphone with a strong Bluetooth antenna like King Kong mini.

Where are your from?

**Comment by pdg212** [comment]
Thanks so much for your prompt reply!Our daughter is also small - she’s 4 years old and using a bulky Medtronic 640g so we don’t want her to have to carry a huge device with her for us to be able to be able to monitor her BG remotely.  I tried a Jelly 2E on Android 12 with the BYOD app mentioned on Reddit but the app kept crashing and the developer could not find any fault after I provided logs from the phone.  If the King Kong 3 mini works well then I’ll get one of those and try again.It’s so, so frustrating that the tech seems to exist but isn’t joined up enough to make a perfect solution.  If the Dexcom G7 receiver had a SIM card (or even a WiFi capability where could use a nearby phone as a hotspot) then we would be very happy.  It feels like it would not have been a huge additional step for Dexcom to do that either. The receiver is really small and light.  What I really can’t understand is how the receiver has a better range.  Whether it is using a different radio channel/frequency with a stronger signal than what is broadcast for the phone?  Very odd!We are in the UK.  Our son is also T1D, both children diagnosed within 2 days of each other during 2020 lockdown.  Our son is a little older and was started on TSlim Control IQ, which works really well. But this is not licensed for our daughter (she’s too young) so we have a very manual setup.  We really want to do better for her as she’s definitely at a disadvantage due to her age :-(Sent from my iPhoneOn 5 Feb 2023, at 20:53, Selcuk Kekec ***@***.***> wrote:﻿
Hi @pdg212,
Unfortunately we had the same problem and tried it with several phones with small dimensions because my son is 6 years old. A big phone is no option to us.
We finally switched to King Kong 3 mini and all problems were gone. The smartphone before, king Kong mini 2 was although terrible. Connection loss and nearly 70 seconds to send a bolus over AAPS by Bluetooth to omnipod dash.
The mini 3 solved all this problems, it is amazing.
Your approach is really interesting. To be honest my knowledge here is limited so I cannot promise you some solution here.
The only advice I can give you is to try out a smartphone with a strong Bluetooth antenna like King Kong mini.
Where are your from?

**Comment by khskekec** [comment]
Hi,

That's really hard to read 😔.

The technical details about the good connection abilities of the stock receiver are at least for me totally hidden.

Also the possibility to read values and function as "man in the middle" with this receiver could be hard and also not sufficient. In general i think it is all about good compatibility of your device and the app and also the Bluetooth antenna. The sensors antenna is totally crap why it has to be compensated from the phone.

The Jelly 2 was really nice with G6 but the G6 was a monster compared to G7's Bluetooth connection. We stopped using the Jelly after 2 days with G7.

If your little princess does not have to wear the phone but it is positinioned in the room only then the form factor is not important for you right?! In this case I would try to find a phone with strong Bluetooth abilities. I can check that if you want.

Unfortunately there is no possibility to put a external Bluetooth dongle in your phone to extend its range.

Did you try other phones also?

My experiences are these:

Samsung Z Flip 4: works really fine but had a broken display after first drop

Jelly 2: bad experience

Google Pixel 6a: Very cheap, really good experience

King Kong 2 mini: crap

King Kong 3 mini: strong Bluetooth and really small dimensions

Kind regards

Selçuk

**Comment by pdg212** [comment]
Hi Selçuk,Thanks again for your reply.  Our daughter carries the phone in a pouch bag with her pump at break time and lunchtime play at school where she is out on the school yard, so a small size is better.  There’s generally no one adult taking care of her during those times so we keep an eye using Dexcom Follow at home.  This worked well with a small iPhone SE and the G6, but the device is a first generation SE (~2018?) and therefore not supported by the Dexcom G7 official app and there does not seem to be a Bring Your Own Dexcom App equivalent for iOS to override the hardware check.  We bought a second hand older Pixel 4a (which is officially supported by G7 app) but not knowing its history we don’t know if the Bluetooth is performing as it should or if it’s damaged in some way. All we know is it does not connect very well with the G7.  I’ve also tried with iPhone 11 and although the initial connection seemed to be quicker, the range is still very, very poor and regularly drops.I’ve ordered the King Kong Mini 3 tonight and it should arrive tomorrow, so will give that a try with the BYOD app and keep fingers crossed for a good result!  If this doesn’t work then I would be willing to try a bigger device with super strong Bluetooth…. else might have to simply revert back to G6 for now, but feels like a step backwards :-(Thanks so much for taking the time to reply back.Best wishes to you and your family,PaulSent from my iPhoneOn 5 Feb 2023, at 22:07, Selcuk Kekec ***@***.***> wrote:﻿
Hi,
That's really hard to read 😔.
The technical details about the good connection abilities of the stock receiver are at least for me totally hidden.
Also the possibility to read values and function as "man in the middle" with this receiver could be hard and also not sufficient. In general i think it is all about good compatibility of your device and the app and also the Bluetooth antenna. The sensors antenna is totally crap why it has to be compensated from the phone.
The Jelly 2 was really nice with G6 but the G6 was a monster compared to G7's Bluetooth connection. We stopped using the Jelly after 2 days with G7.
If your little princess does not have to wear the phone but it is positinioned in the room only then the form factor is not important for you right?! In this case I would try to find a phone with strong Bluetooth abilities. I can check that if you want.
Unfortunately there is no possibility to put a external Bluetooth dongle in your phone to extend its range.
Did you try other phones also?
My experiences are these:
Samsung Z Flip 4: works really fine but had a broken display after first drop
Jelly 2: bad experience
Google Pixel 6a: Very cheap, really good experience
King Kong 2 mini: crap
King Kong 3 mini: strong Bluetooth and really small dimensions
Kind regards
Selçuk

**Comment by khskekec** [comment]
Hi Paul,

In this case size matters and the King Kong Mini 3 might be a good tradeoff.

I hope that you will make the same positive experience as me and it will be not like another wrong investment.

Please keep me up 2 date.

Additionaly you could think about trying out android AAPS with an omnipod pump to collect first important experience.

I feel with you. It was horrible for us also to manage diabetes manually for our son. Nearly 2,5 years without sleep was a torture but AAPS brought us the solution we were waiting for.

Kind regards

Selçuk

**Comment by pdg212** [comment]
Hi Selçuk,The King Kong 3 Mini arrived and I managed to setup the BYOD app with no issues and it runs stable without crashing (unlike the Jelly 2E), presumably due to better hardware.  The device is similar in size to her old iPhone SE (~2018 model) so that’s also good.The Bluetooth connectivity does seem to be stronger than the Pixel 4a… though does need to be very close by and certainly a lot closer than the iPhone SE for G6 (which works from another room) or the G7 receiver (which also has good range).We started a new G7 session yesterday evening.  In the meantime she is also wearing a G6 which has been on her for a few days… so we could continue to get some readings remotely while she’s in school.  I have not calibrated either sensor but as you’ll see from this snippet, the readings are quite different, though the “shape” is similar.  G7 is reporting 1-2mmol/L higher than her finger prick reading.  G6 was closer to finger but the spike at approx 3am neither were right… she was actually 14 on finger when G7 thought 16 and G6 thought 11.  Will wait for 24 hours as I’ve read that the first 24 hours can be unstable, then consider a calibration… though we generally never calibrated G6.Do you find you need to calibrate G7, even if it’s only once per session near the beginning?I experimented also with a G7 on me (I’m not diabetic) and it was on average 1 mmol/L above actual finger prick reading, so after 30 hours and a fasting test this morning I have calibrated.  My daughter’s first G7 was also reading higher than reality before calibration which is a concern as it may lead to false sense of security (I.e. Dexcom showing 4 and she’s really 3). She has no hypo awareness.  Be interesting to know your experience.I am curious about trying the AAPS / Loop3 but also wondering if it’ll be better to wait for Omnipod 5…. Is there a lot of work to constantly tweak it?Kind regards,PaulSent from my iPhoneOn 6 Feb 2023, at 14:34, Selcuk Kekec ***@***.***> wrote:﻿
Hi Paul,
In this case size matters and the King Kong Mini 3 might be a good tradeoff.
I hope that you will make the same positive experience as me and it will be not like another wrong investment.
Please keep me up 2 date.
Additionaly you could think about trying out android AAPS with an omnipod pump to collect first important experience.
I feel with you. It was horrible for us also to manage diabetes manually for our son. Nearly 2,5 years without sleep was a torture but AAPS brought us the solution we were waiting for.
Kind regards
Selçuk

**Comment by khskekec** [comment]
Hi Paul,

Nice that you are already using the King Kong. I hope it worked well so far.

Regarding your calibration issue:

We have extreme lows right after the warm up four quite long. Even in this case we do not calibrate. Sometimes at the end of its lifetime the G7 starts again to report jumpy values and also higher ones compared to blood measurements. These cases are recurring all the time but we are now used to it.

My advice :

Try to find out the best way for you and the little princess.

I also heard about a update for the G7 with improved accuracy. Might be that this will solve our problems.

Regarding your question to omnipod 5:

I have no information about the current state of development for omnipod 5.

Kind regards

Selçuk
